### ISSUE TITLE
Pre-configure reoute builders at build time

### ISSUE THREAD

**Issue by lburgazzoli** [apache/camel-quarkus#511]
As today we do not perform so may optimizations and introspection to the routes so we may introduce a new annotation to give some hints to the build processor about in which phase a route is expected to be configured and some additional meat-data like classes that need to be registered for reflection, like:

```java
import org.apache.camel.quarkus.annotations.Configure;
import io.quarkus.deployment.annotations.ExecutionTime;

@Configure(ExecutionTime.STATIC_INIT, reflectiveClasses = { "com.acme.MyBean" })
public class MyRouteBuilder extends RouteBuilder {
    @Override
    public void configure() {
        from("somewhere")
            .bean("com.acme.MyBean", "doSomething");
    }
}
```

toughs ?

**Comment by ppalaga** [comment]
Related https://github.com/apache/camel-quarkus/issues/97 and partly also https://github.com/apache/camel-quarkus/issues/253 

Yes, this might help to solve the problem, although hard to say how reliably we can pre-configure a route in this way without knowing its complete model at build time. 
I am asking myself, whether we would still need this kind of annotation if we had the complete Route model (which we do not ATM due to how RoutesBuilder is designed) at build time. We could perhaps figure out many of the necessary parts, but probably not all: e.g. a Route definition can contain an ad hoc lambda, that would be impossible to introspect at build time. So we'd perhaps still need some way (an annotation or an app.props entry) to pass the info to the Processor.

**Comment by lburgazzoli** [comment]
Yep, this is mainly and hint a user can give to the builder and it is under the user responsibility so if it fails, well, we'll let the user figure out how to fix it.

In any case it does not prevent us to do more introspection when we'll have the capabilities to do it so maybe in the future such annotation can be deprecated/removed

**Comment by ppalaga** [comment]
BTW do you think the effort needed to support this (and the new potential bugs) is in balance with the boot time improvements? In other words are our boot times bad enough to invest here instead of elsewhere? (My knowledge in this area is limited to running rather simple demos so I frankly do not know whether we generally need to improve the boot times).

**Comment by lburgazzoli** [comment]
I think we need to experiment a little bit and we can mark this annotation/feature as experimental and drop it if it does not make any sense

**Comment by lburgazzoli** [comment]
The `reflectiveClasses` field is not needed as this cases is covered already bythe quarkus annotation [RegisterForReflection](https://github.com/quarkusio/quarkus/blob/1.0/core/runtime/src/main/java/io/quarkus/runtime/annotations/RegisterForReflection.java#L32)
